### PR TITLE
feat: TypeScript 4.8+ support and NativeClass decorator improvements

### PR DESCRIPTION
**Pull request by NathanWalker**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/DevelopmentWorkflow.md#running-unit-tests-application.

## What is the current behavior?

TypeScript 4.8+ could not be used.

## What is the new behavior?

TypeScript 4.8+ can now be used.

closes https://github.com/NativeScript/NativeScript/issues/10024

### CODE REVIEW

File: packages/webpack5/src/transformers/NativeClass/index.ts [hunk 1]
```diff
@@ -24,14 +25,14 @@ export default function (ctx: ts.TransformationContext) {
...
-		return ts.createIdentifier(
+		return ts.factory.createIdentifier(
 			ts
 				.transpileModule(
 					node.getText().replace(/@NativeClass(\((.|\n)*?\))?/gm, ''),
 					{
 						compilerOptions: {
 							noEmitHelpers: true,
-							module: ts.ModuleKind.CommonJS,
+							module: ts.ModuleKind.ESNext,
```
Comment (rigor789) [inline packages/webpack5/src/transformers/NativeClass/index.ts:35]: Is this intentional? I believe we need to compile the class as CommonJS for the native class extension to work...

Comment (NathanWalker) [reply packages/webpack5/src/transformers/NativeClass/index.ts:35]: The module can be `esnext` - it's the `target` that matters - next line stays es5.

Comment (edusperoni) [reply packages/webpack5/src/transformers/NativeClass/index.ts:35]: the CommonJS vs ESNext Module is just a matter of:
```
var something = class here;
exports.something = something
```

versus:

```
var something = class here;
export {something};
```

Comment (rigor789) [reply packages/webpack5/src/transformers/NativeClass/index.ts:35]: Makes total sense 👍 Thanks for clarifying!

**Pull request merged**